src/main/java/com/useful/ucars/uCarsListener.java

```diff
@@ -553,22 +560,23 @@ public void tickCalcsAndLegacy(VehicleUpdateEvent event) {
 		if(event instanceof ucarUpdateEvent){
 			travel = ((ucarUpdateEvent) event).getTravelVector().clone();
 		}
-
 		if(!(event instanceof ucarUpdateEvent)){ //If it's just the standard every tick vehicle update event...
 			if(UEntityMeta.hasMetadata(vehicle, "car.vec")){ //If it has the 'car.vec' meta, we need to use RACE CONTROLS on this vehicle
 				ucarUpdateEvent evt = (ucarUpdateEvent) UEntityMeta.getMetadata(vehicle, "car.vec").get(0).value(); //Handle the update event (Called here not directly because otherwise ppl with a better connection fire more control events and move marginally faster)
 				evt.player = ((Player)passenger); //Set the player (in the car) onto the event so it can be handled by uCarUpdate handlers
 				evt.incrementRead(); //Register that the control input update has been executed (So if no new control input event within 2 ticks; we know to stop the car)
 				UEntityMeta.removeMetadata(vehicle, "car.vec"); //Update the 'car.vec' metadata with an otherwise identical event; but without the player object attached
-				ucarUpdateEvent et = new ucarUpdateEvent(vehicle, evt.getTravelVector().clone(), null, evt.getDir()); //Clone of the other event, except no player object attached
+				ucarUpdateEvent et = new ucarUpdateEvent(vehicle, evt.getTravelVector().clone(), evt.getPlayer(), evt.getDir()); //Clone of the other event, except no player object attached
```

Comment:
The player shouldn't be attached to the event here as I think that is a potential memory leak? (It was null before I think so that a car in the map isn't keeping player objects alive that ought to have been cleaned up.)

Reply:
I think I added that for testing and forgot to remove it